### PR TITLE
[DB-11882]: Add a caveat or warning in oracle assessment report about Partition being colocated by default

### PR DESCRIPTION
**Pull request by sanyamsinghal**
https://yugabyte.atlassian.net/browse/DB-11882

### CODE REVIEW

File: yb-voyager/cmd/assessMigrationCommand.go [hunk 1]
```diff
@@ -853,6 +856,39 @@ func fetchColumnsWithUnsupportedDataTypes() ([]utils.TableColumnsDataTypes, erro
...
+const ORACLE_PARTITION_DEFAULT_COLOCATION = `In case of Oracle, all partitions of a partitioned table are created as colocated by default. 
+The Assessment Report provides sharding/colocation recommendations for each partition individually.<br>
+However, due to certain limitations, these recommendations cannot be directly applied. 
+To manually modify the schema for sharding, please refer: <a class="highlight-link" href="https://github.com/yugabyte/yb-voyager/issues/1581">https://github.com/yugabyte/yb-voyager/issues/1581</a>.`
+
+func addNotesToAssessmentReport() {
+	log.Infof("adding notes to assessment report")
+	switch source.DBType {
+	case ORACLE:
+		assessmentReport.Notes = append(assessmentReport.Notes, ORACLE_PARTITION_DEFAULT_COLOCATION)
```
Comment (makalaaneesh) [inline yb-voyager/cmd/assessMigrationCommand.go:868]: let's only add this if there are actual partitioned tables in the schema.

File: yb-voyager/cmd/templates/assessmentReport.template [hunk 2]
```diff
@@ -176,6 +186,17 @@
         {{if not $hasUnsupportedFeatures}} <!-- Check if no unsupported features were found -->
             <p>No unsupported features were detected among the ones assessed.</p>
         {{end}}
+
+        <div class="notes-box">
+            <h4>Notes</h4>
+            {{if .Notes}}
```
Comment (makalaaneesh) [inline yb-voyager/cmd/templates/assessmentReport.template:192]: let's have the if condition at the div level. 
I.e. if Notes is empty, let's not even have the div, heading, etc.

File: yb-voyager/cmd/assessMigrationCommand.go (hunk 1 above)
Comment (sanyamsinghal) [reply yb-voyager/cmd/assessMigrationCommand.go:868]: hmm makes sense. I will add a check based on whether a `partition.sql` file is present or not

File: yb-voyager/cmd/templates/assessmentReport.template (hunk 2 above)
Comment (sanyamsinghal) [reply yb-voyager/cmd/templates/assessmentReport.template]: changes done, @makalaaneesh

**Review by makalaaneesh — APPROVED** [review]
LGTM. Pls test this manually once with and without partitions to ensure that notes get added only if partitioned tables are present. 

Is it easy to add this to assessment report validation? If so, we should do it. cc: @shubham-yb

**Comment by sanyamsinghal** [comment]
> Is it easy to add this to assessment report validation? If so, we should do it. cc: @shubham-yb
@makalaaneesh  yes it should be easily parseable from JSON report. @shubham-yb let me know if you need any help

**Pull request merged**